internal/ingress/controller/kong.go

```diff
@@ -247,7 +247,12 @@ func (n *NGINXController) syncServices(ingressCfg *ingress.Configuration) (bool,
 				}
 
 				if s != nil {
+					// if kongIngress object does not exist, don't create the service in Kong
+					outOfSync := false
 					if kongIngress != nil && kongIngress.Proxy != nil {
+						// check if we need to patch KongService later
+						outOfSync = !reflect.DeepEqual(s, kongIngress.Proxy)
```

Comment:
The comparison seems flawed. Have you tested this?
`kongadminv1.Service` and `Proxy` are two different objects with `Service` containing fields which are not present in `Proxy` struct.

It will be better to write a separate method comparing the two objects field by field.
Also, think about cases where, the fields are populated in `s` but not `in `kongIngress.Proxy`, which results in a `PATCH` when not needed.

Some helpful context here: `kongIngress.Proxy` is used to override properties defined in `s` only.  If fields are absent in `kongIngress.Proxy`, then the value in `s` take precedence.